src/GitExtensions.Extensibility/build/net461/GitExtensions.Extensibility.targets

```diff
@@ -1,24 +1,37 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <!-- The following properties may be overwritten by plugins. In case they are not defined, they assume default values. -->
+    <PropertyGroup>
+        <GitExtensionsDownloadPath Condition="$(GitExtensionsDownloadPath) == ''">..\..\gitextensions.shared</GitExtensionsDownloadPath> <!-- path is relative to $(ProjectDir) -->
+        <GitExtensionsReferenceVersion Condition="$(GitExtensionsReferenceVersion) == ''">latest</GitExtensionsReferenceVersion> <!-- 'latest' or 'v3.1' (= tag from GitHub releases) or 'v3.1.0.5877' (= build number from AppVeyor)-->
+        <GitExtensionsReferenceSource Condition="$(GitExtensionsReferenceSource) == ''">GitHub</GitExtensionsReferenceSource> <!-- 'GitHub' or 'AppYevor' -->
+        <GitExtensionsPath Condition="$(GitExtensionsPath) == ''">$([System.IO.Path]::Combine('$(_GitExtensionsDownloadPath)', 'GitExtensions'))</GitExtensionsPath> <!-- for local builds (no download) -->

```

Comment:
These some story about `.props` and `.targets`, if I remember it `.props` are included at the top of the project and `.targets` at the bottom. It also plays a role when user tries to use them inside the `.csproj`.

Reply:
I spend quite a long time with it when I was making the package to work, also with testing various setups.

Reply:
I think in my case the problem was the order of properties. In a previous version I wrote (and tested):
```
$_GitExtensionsDownloadPath = $ProjectDir + $GitExtensionsDownloadPath
$GitExtensionsPath = $_GitExtensionsDownloadPath + 'GitExtensions'
```

Then I decided to move `$GitExtensionsPath` within the same file upwards, so I got:
```
$GitExtensionsPath = $_GitExtensionsDownloadPath + 'GitExtensions'
$_GitExtensionsDownloadPath = $ProjectDir + $GitExtensionsDownloadPath
```
And this failed because  `$GitExtensionsPath` depends on `$_GitExtensionsDownloadPath`, which is only defined later.

I now changed it to:
```
$GitExtensionsPath = $ProjectDir + $GitExtensionsDownloadPath + 'GitExtensions'
$_GitExtensionsDownloadPath = $ProjectDir + $GitExtensionsDownloadPath
```
To make `$GitExtensionsPath` from `$_GitExtensionsDownloadPath` and tested it again.

Reply:
Silly question, why did you moved it?

Reply:
> Silly question, why did you moved it?

To have all properties grouped together rather than distributed in two different files.
> Is `_GitExtensionsDownloadPath ` used anymore?

This PR doesn't change anything in this regard. It is still the absolute path to the download dir and used in the prebuild event and passed as parameter to the PS script.

Reply:
Ok